Assets/AWSIM/Scenes/Main/PointCloudMapping/Scripts/Mapper/RGLMappingAdapter.cs

```diff
@@ -29,43 +45,46 @@ public void Start()
             // Make sure automatic capture in RGL Lidar Sensor is disabled.
             // We want to perform captures only on demand (after warping).
             lidarSensor.AutomaticCaptureHz = 0;
+
+            if (leafSize > 0.0f)
+            {
+                rglGraphMapping.UpdateNodePointsDownsample(downsampleNodeId, new Vector3(leafSize, leafSize, leafSize));
+            }
+            else
+            {
+                rglGraphMapping.RemoveNode(downsampleNodeId);
+            }
+
+            lidarSensor.ConnectToWorldFrame(rglGraphMapping);
         }
 
         public string GetSensorName()
         {
             return gameObject.name;
         }
 
-        public PointCloudOfXYZI Capture_XYZI_ROS(Vector3 worldOriginROS)
+        public void SetOutputPcdFilePath(string path)
+        {
+            outputPCDFilePath = path;
+            rglGraphMapping.UpdateNodePointsWritePCDFile(writePcdNodeId, outputPCDFilePath);
+        }
```

Comment:
This was a temporary implementation for one mapping sensor at the same time. I am aware that it will be changed when extending to multisensor mapping. Anyway, I applied your solution proposition.